buildcontext/gitlookup.go

```diff
@@ -230,11 +231,24 @@ func (gl *GitLookup) detectProtocol(host string) (protocol gitProtocol, err erro
 		Auth: []ssh.AuthMethod{
 			ssh.PublicKeysCallback(agent.NewClient(sshAgent).Signers),
 		},
+		HostKeyAlgorithms: []string{
```

Comment:
So to clarify, this list is the same as supportedHostKeyAlgos but with `ssh.KeyAlgoRSA` first for compatibility?

Reply:
yeah, I took the entire list from `list from golang.org/x/crypto/ssh/common.go supportedHostKeyAlgos` and moved `ssh.KeyAlgoRSA` to the top.

Reply:
The list seems different from the one referenced here? https://cs.opensource.google/go/x/crypto/+/ae814b36:ssh/common.go;l=69-81;bpv=0;bpt=0. Could it have changed in a newer version or something?

Reply:
ah ha! yeah mine came from `mod/golang.org/x/crypto@v0.0.0-20200820211705-5c72a883971a/ssh/common.go`.

Ultimately I'm not sure it matters that much in the context of this PR, as we just need to have `ssh.KeyAlgoRSA` at the top.

When we loop back and allow support for multiple pre-configured key scans, I want to remove this list.

Reply:
sgtm, thanks!